quickwit-index-config/src/config.rs

```diff
@@ -70,17 +70,24 @@ pub trait IndexConfig: Send + Sync + Debug + DynClone + 'static {
     /// Returns the document built from a json string.
     fn doc_from_json(&self, doc_json: &str) -> Result<Document, DocParsingError>;
     /// Returns the schema.
+    ///
+    /// Considering schema evolution, splits within an index can have different schema
+    /// over time. The schema returned here represents the most up-to-date schema of the index.
     fn schema(&self) -> Schema;
     /// Returns the query.
-    fn query(&self, request: &SearchRequest) -> Result<Box<dyn Query>, QueryParserError>;
+    fn query(
+        &self,
+        schema: Schema,
```

Comment:
Rename split_schema and document that this is the schema that is specific to the targetted split, as opposed to current schema.